Add tt CLI docs (#2973)
tarantool/doc: doc/reference/tt_cli/configuration.rst

tt CLI configuration
====================

.. _tt-config_file:

Configuration file
------------------

> The ``tt`` configuration file defines aspects of its runtime, such as directories

Reviewer: The configuration file defines the configuration of the current environment. It defines (will define) not only the runtime instance. It's easier to talk with your voice.

that it uses and log retention parameters.

By default, the configuration file is called ``tarantool.yaml``. The location
where ``tt`` searches it depends on the :ref:`launch mode <tt-config_modes>`.
You can also pass the configuration file explicitly in the ``--cfg``
:doc:`argument <arguments>`.

``tt`` configuration file is a YAML file with the following content:

..  code:: yaml

    tt:
        modules:
            directory: path/to/modules/dir
        app:
            instances_available: path/to/available/applications
            run_dir: path/to/run_dir
            log_dir: path/to/log_dir
            log_maxsize: num (MB)
            log_maxage: num (Days)
            log_maxbackups: num
            restart_on_failure: bool

modules section
~~~~~~~~~~~~~~~

* ``directory`` -- a directory where external modules are stored.

app section
~~~~~~~~~~~

*   ``instances_available`` -- a directory where :ref:`instances <admin-instance_file>`
    are stored.
*   ``run_dir``-- a directory for instance runtime artifacts, such as console
    sockets or PID files.

> *   ``log_dir`` -- a directory where log files are stored.

Author: @LeonidVas could you help with the default values?

Reviewer: By default the current directory will be used, but it will be changed soon.

> tt CLI configuration

Reviewer 2: ```suggestion
Configuration
```
Same as with the installation page. The user sees in the tree on the left that it's an instruction for tt CLI, so duplicating the name is not necessary.

====================

.. _tt-config_file:

Configuration file
------------------

The key artifact that defines the ``tt`` environment and various aspects of its
execution is its configuration file.

By default, the configuration file is called ``tarantool.yaml``. The location

> where ``tt`` searches it depends on the :ref:`launch mode <tt-config_modes>`.

Reviewer 2: ```suggestion
where ``tt`` searches for it depends on the :ref:`launch mode <tt-config_modes>`.
```

You can also pass the configuration file explicitly in the ``--cfg``
:doc:`argument <arguments>`.

> ``tt`` configuration file is a YAML file with the following content:

Reviewer 2: ```suggestion
The ``tt`` configuration file is a YAML file with the following content:
```

..  code:: yaml

    tt:
        modules:
            directory: path/to/modules/dir
        app:
            instances_available: path/to/available/applications
            run_dir: path/to/run_dir
            log_dir: path/to/log_dir
            log_maxsize: num (MB)

>             log_maxage: num (Days)

Reviewer 2: ```suggestion
            log_maxage: num (days)
```

            log_maxbackups: num
            restart_on_failure: bool

modules section
~~~~~~~~~~~~~~~

> * ``directory`` -- a directory where external modules are stored.

Reviewer 2: ```suggestion
* ``directory`` -- the directory where external modules are stored.
```
It's like "the specific" directory. External modules are not stored in any other directories, there's only one.

.. // TODO: add link to external modules doc page when it's ready

app section
~~~~~~~~~~~

*   ``instances_available`` -- a directory where :ref:`instances <admin-instance_file>`
    are stored.
*   ``run_dir``-- a directory for instance runtime artifacts, such as console
    sockets or PID files.

> *   ``log_dir`` -- a directory where log files are stored.

Reviewer 2: ```suggestion
*   ``instances_available`` -- the directory where :ref:`instances <admin-instance_file>`
    are stored.
*   ``run_dir``-- the directory for instance runtime artifacts, such as console
    sockets or PID files.
*   ``log_dir`` -- the directory where log files are stored.
```

> *   ``log_maxsize`` -- the maximum size of the log file before it gets rotated,

Reviewer 2: Just a tip: make sure to cross-link to the `logrotate` page from here, once it is created.

    in megabytes. Default: 100.
*   ``log_maxage`` -- the maximum age of log files in days. The age of a log
    file is defined by the timestamp encoded in its name. Default: not defined

>     (don't delete log files based on their age).

Reviewer 2: After a discussion in the messenger, we decided to rephrase this for clarity.

    ..  note::

        A day is defined as exactly 24 hours. It may not exactly correspond to
        calendar days due to daylight savings, leap seconds, and other time adjustments.

*   ``log_maxbackups`` -- the maximum number of stored log files.

>     Default: not defined (don't delete log files based on their count).

Reviewer 2: After a discussion in the messenger, we decided to rephrase this for clarity.

*   ``restart_on_failure`` -- restart the instance on failure: ``true`` or ``false``.
    Default: ``false``.

.. _tt-config_modes:

Launch modes
------------

> ``tt`` launch mode defines its working directory and the way it searches the configuration file:

Reviewer 2: ```suggestion
``tt`` launch mode defines its working directory and the way it searches for the configuration file:
```

..  container:: table

    ..  list-table::

>         :widths: 10 25 35 30

Reviewer 2: This is how it looks now on smaller laptop screens used by many developers:

![image](https://user-images.githubusercontent.com/40490252/177938764-3cc898de-b238-49aa-86b5-c8092f36f661.png)

Unfortunately, this is a limitation of our site design.
I propose listing the launch modes and then creating a subsection for each of them (not necessarily a table). In this way, the text in the cells won't break. It's especially important that console options, like `--local=DIRECTORY`, don't break in the middle.

        :header-rows: 1

        *   -   Mode
            -   Argument
            -   Configuration file location
            -   Working directory
        *   -   Default
            -   --
            -   Searched from the current directory to the root.
                ``/etc/tarantool`` if the file is not found.
            -   The directory where the configuration file is found.
        *   -   System launch
            -   ``--system``

                ``-S``
            -   ``/etc/tarantool``

>             -   Current directory

Reviewer 2: ```suggestion
            -   Current directory.
```
Consistency

        *   -   Local launch
            -   ``--local=DIRECTORY``

                ``-L=DIRECTORY``
            -   Searched from the specified directory to the root.
                ``/etc/tarantool`` if the file is not found.
            -   The specified directory.

>                 If tarantool or tt executable files are found in working directory,
>                 they will be used further.

Reviewer 2: ```suggestion
                If tarantool or tt executable files are found in the working directory,
                they will be used.
```
